### PR TITLE
Add Eslint Configuration & Fix All issues

### PR DESCRIPTION
**Pull request by Yedidyar**
it pretty big i tried to reduce the cognitive complexity but most  of the + is from the lock file

### CODE REVIEW

File: src/report.ts [hunk 1]
```diff
@@ -17,12 +17,12 @@ export const logReport = (
 		// No table matching found
 
 		p.log.info(
-			`${green(`Type need to be manually typed`)}:
-  ${yellow(report.typeCouldBeBetter.map((c) => `${c}`).join("\n  "))}`
+			`${green("Type need to be manually typed")}:
```
Comment (jycouet) [inline src/report.ts:20]: Could we tweak this? 

I like a lot `'` as you can add variables inside. With " you always have to change things...

**Review by jycouet — CHANGES_REQUESTED** [review]
The rest looks good 👍
nothing broke

File: src/report.ts (hunk 1 above)
Comment (jycouet) [reply src/report.ts:20]: (I mean the backtick)

Comment (Yedidyar) [reply src/report.ts:20]: sonar lint complainant about that but i can disable this rule 
what do you prefer?

**Pull request merged**